### DISCUSSION TITLE
Add unread and mention count properties and events to `Channel` class

### DISCUSSION THREAD

**Discussion by jai-x** [ppy/osu#18564] [Ideas]
In the current state:
Unread state of a channel is tracked via the `LastReadId` message being compared against messages currently present in the channel.
https://github.com/ppy/osu/blob/4d0c3fcee28d93b871948982168586c0eb8b5e15/osu.Game/Online/Chat/Channel.cs#L43
Mentions are tracked externally from the class via the `MessageNotifier` class which only posts a notification on mentions or DMs:
https://github.com/ppy/osu/blob/master/osu.Game/Online/Chat/MessageNotifier.cs

I think it would be an improvement to have the unread state and the mention count be added as properties to the `Channel` class, either using `Bindable`s or events such that it becomes easier for chat overlay components to display this state for the channels.

I'm now quite sure the best place to put code to update such values on the class. Maybe `MessageNotifier` should increase in scope or the `Channel` class itself should track these values. Looking for feedback on this idea.

**Comment by peppy** [comment]
Out of curiosity, have you looked at the newer websocket based chat API, and how the interaction flow is for that? We'd want to look at switching to that as a priority, because the current one causes very expensive database queries each poll for certain users.

**Reply by jai-x** [reply]
I haven't taken a look at that at all? Is there any documentation available for that anywhere?

**Reply by peppy** [reply]
As far as I know there is none. cc/ @notbakaneko
